### PR TITLE
[Backtracing][Frontend] Add support for implicit import of _Backtracing

### PR DESCRIPTION
**Pull request by al45tair**
Once the API has gone through Swift Evolution, we will want to implicitly import the _Backtracing module.  Add code to do that, but set it to off by default for now.

rdar://105394140

### CODE REVIEW

**Comment by al45tair** [comment]
This is part of a set of stacked PRs:

|  | PR                | Title                                                                      | Merges Into       |
|--|:------------------|:---------------------------------------------------------------------------|:------------------|
|  | apple/swift#63650 | [Backtracing] Add support for building target executables into libexec     | main              |
|  | al45tair/swift#8  | [Backtracing] Add support for looking up paths for auxiliary executables   | apple/swift#63650 |
|  | al45tair/swift#9  | [Backtracing][Runtime][Darwin] Add code to trigger the external backtracer | al45tair/swift#8  |
|  | al45tair/swift#10 | [Backtracing] Add _Backtracing library to the build                        | al45tair/swift#9  |
|👉| al45tair/swift#11 | [Backtracing][Frontend] Add support for implicit import of _Backtracing    | al45tair/swift#10 |
|  | al45tair/swift#12 | [Backtracing] Add swift-backtrace to the build                             | al45tair/swift#11 |
|  | al45tair/swift#13 | [Backtracing][Docs] Add an explanation of the workings of the backtracer   | al45tair/swift#12 |
|  | al45tair/swift#14 | [Backtracing] Add control over symbol caching                              | al45tair/swift#13 |

(Only the bottom PR will be able to run PR checks at any time. Please don't wait for the checks before review.)

File: cmake/modules/SwiftImplicitImport.cmake [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Test if the Swift compiler supports -disable-implicit-<module>-module-import
+function(swift_supports_implicit_module module_name out_var)
+  file(WRITE "${CMAKE_BINARY_DIR}/tmp/empty-check-${module_name}.swift" "")
```
Comment (kavon) [inline cmake/modules/SwiftImplicitImport.cmake:3]: nice 👍🏼

File: lib/AST/NameLookup.cpp [hunk 2]
```diff
@@ -662,6 +662,22 @@ static void recordShadowedDeclsAfterTypeMatch(
         }
       }
 
+      // Next, prefer any other module over the _Backtracing module.
+      if (auto spModule = ctx.getLoadedModule(ctx.Id_Backtracing)) {
+        if ((firstModule == spModule) != (secondModule == spModule)) {
+          // If second module is _StringProcessing, then it is shadowed by
```
Comment (kavon) [inline lib/AST/NameLookup.cpp:668]: This is right. Would be nice if there was a little local function to clean up some of the duplicated parts of this check-and-shadow logic for each module.

**Review by kavon — APPROVED** [review]
LGTM 🥳

**Pull request closed without merging**